docs/autoapi_templates/index.rst

```diff
@@ -0,0 +1,130 @@
+..  Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+..    http://www.apache.org/licenses/LICENSE-2.0
+
+..  Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+API Reference
+=============
+
+Operators
+---------
+Operators allow for generation of certain types of tasks that become nodes in
+the DAG when instantiated. All operators derive from :class:`~airflow.models.BaseOperator` and
+inherit many attributes and methods that way.
+
+There are 3 main types of operators:
+
+- Operators that performs an **action**, or tell another system to
+  perform an action
+- **Transfer** operators move data from one system to another
+- **Sensors** are a certain type of operator that will keep running until a
+  certain criterion is met. Examples include a specific file landing in HDFS or
+  S3, a partition appearing in Hive, or a specific time of the day. Sensors
+  are derived from :class:`~airflow.sensors.base_sensor_operator.BaseSensorOperator` and run a poke
+  method at a specified :attr:`~airflow.sensors.base_sensor_operator.BaseSensorOperator.poke_interval` until it returns ``True``.
+
+BaseOperator
+''''''''''''
+All operators are derived from :class:`~airflow.models.BaseOperator` and acquire much
+functionality through inheritance. Since this is the core of the engine,
+it's worth taking the time to understand the parameters of :class:`~airflow.models.BaseOperator`
+to understand the primitive features that can be leveraged in your
+DAGs.
+
+BaseSensorOperator
+''''''''''''''''''
+All sensors are derived from :class:`~airflow.sensors.base_sensor_operator.BaseSensorOperator`. All sensors inherit
+the :attr:`~airflow.sensors.base_sensor_operator.BaseSensorOperator.timeout` and :attr:`~airflow.sensors.base_sensor_operator.BaseSensorOperator.poke_interval` on top of the :class:`~airflow.models.BaseOperator`
+attributes.
+
+Operators packages
+''''''''''''''''''
+All operators are in the following packages:
+
+.. toctree::
+  :includehidden:
+  :glob:
+  :maxdepth: 1
+
+  airflow/operators/index
```

Comment:
General comment about  this file: This doesn't feel like a "template" - is this just how autoapi works?

Could these instead be 
```
:module:\`airflow.operators`
```

And then this index  file commited as a "normal" rst file under `docs/autoapi/index.rst` (Though see further comment about url path.

Reply:
This is a template to keep the old look and make the documentation clearer.
Original file: https://raw.githubusercontent.com/rtfd/sphinx-autoapi/master/autoapi/templates/index.rst

Autoapi works in such a way that it analyzes the source code and generates rst files based on the template.  This is done before the actual sphinx process starts.

Reply:
Oh right yes, so  we do need this template then. Cool